core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -366,6 +376,31 @@ contract PricelessPositionManager is FeePayer {
         emit SettleExpiredPosition(msg.sender, totalRedeemableCollateral.rawValue, tokensToRedeem.rawValue);
     }
 
+    /**
+    * @notice Premature contract settlement under emergency circumstances.
+    * @dev Only the governor can call this function as they are permissioned within the `FinancialContractAdmin`.
+    * Upon emergency shutdown, the contract settlement time is set to the shutdown time. This enables withdrawal
+    * to occur via the standard settleExpired function call. Contract state is set to `ExpiredPriceRequested`
+    * which prevents re-entreaty into this function or the `expire` function.
+    */
+    function emergencyShutdown() external onlyPreExpiration() onlyOpenState() {
+        require(msg.sender == _getFinancialContractsAdminAddress());
+
+        contractState = ContractState.ExpiredPriceRequested;
+        // Expiratory time now becomes the current time (emergency shutdown time).
+        // Price requested at this time stamp. `settleExpired` can now withdraw at this timestamp.
+        uint oldExpirationTimestamp = expirationTimestamp;
+        expirationTimestamp = getCurrentTime();
+        _requestOraclePrice(expirationTimestamp);
+
+        emit EmergencyShutdown(msg.sender, oldExpirationTimestamp, expirationTimestamp);
+    }
+
+    //TODO: is this how we want this function to be implemented?
+    function remargin() external onlyPreExpiration() {
+        _payFinalFees(address(this));
```

Comment:
The function you want is `payFees()`. However, there are cases where payFees returns some collateral to the caller, and there's not way to extract that money from the administrator if it were to call it. I'm not exactly sure what to do about that. 

Maybe we should make this function do nothing? If we were to do that, it would violate our official rules around what remargin is supposed to do (according to our docs). Maybe we should change the definition of the remargin function to not require that fees be paid if no money moves, etc.

Reply:
ok so for now I will leave the todo and make the function do nothing